examples/DeviceASAN.cpp

```diff
@@ -0,0 +1,25 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2024, Lawrence Livermore National Security, LLC and CARE
+// project contributors. See the CARE LICENSE file for details.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//////////////////////////////////////////////////////////////////////////////
+
+// CARE library header
+#include "care/care.h"
+
+int main(int, char**) {
+   const int size = 1000000;
+   care::host_device_ptr<int> data(size);
+
+   CARE_STREAM_LOOP(i, 0, size) {
+      data[i] = i;
+   } CARE_STREAM_LOOP_END
+
+   CARE_STREAM_LOOP(i, 0, size + 1) {
```

Comment:
I would just add a comment about how this is an intentional bug going one past the end, and also describe what the expected address sanitizer output should look like in response to the intentional bug.

Reply:
Good idea! Done.

Reply:
Thanks!

Reply:
I don't see the comment, don't forget to commit it before you merge.